plugins/opensslcryptoplugin/evp/evp.c

```diff
@@ -240,6 +240,103 @@ int osslevp_aes_decrypt_ciphertext(const EVP_CIPHER *evp_cipher,
     return plaintext_length;
 }
 
+int osslevp_aes_auth_encrypt_plaintext(const EVP_CIPHER *evp_cipher,
+                                       const unsigned char *init_vector,
+                                       const unsigned char *key,
+                                       int key_length,
+                                       const unsigned char *auth,
+                                       int auth_length,
+                                       const unsigned char *plaintext,
+                                       int plaintext_length,
+                                       unsigned char **encrypted,
+                                       unsigned char **tag,
+                                       int tag_length)
+{
+    int ciphertext_length = plaintext_length + AES_BLOCK_SIZE;
+    int update_length = 0;
+    int final_length = 0;
+    unsigned char *ciphertext = NULL;
+    unsigned char *tag_output = NULL;
+
+    if (evp_cipher == NULL || plaintext_length <= 0 || plaintext == NULL
+            || auth == NULL || auth_length <= 0
+            || key_length <= 0 || key == NULL || encrypted == NULL || tag_length <= 0) {
+        /* Invalid arguments */
+        fprintf(stderr, "%s\n", "invalid arguments, aborting encryption");
+        return -1;
+    }
+
+    /* Allocate the buffer for the encrypted output */
+    ciphertext = (unsigned char *)malloc(ciphertext_length);
+    memset(ciphertext, 0, ciphertext_length);
```

Comment:
Consider using `OPENSSL_zalloc` for this.

Reply:
Yes sounds good; perhaps we could fix this across the codebase in a separate patch.

Reply:
Ok!